### ISSUE TITLE
[Working Group Proposal] Eventing Kafka Working Group

### ISSUE THREAD

**Issue by devguyio** [knative/community#693]
### Motivation/Abstract

Within the knative-sandbox org, Kafka based eventing currently includes 2 channel implementations, 2 sources implementations, 1 broker and 1 sink with more than three vendors and multiple users for production workloads. With that, it is desired to have a dedicated forum for Kafka based Eventing which can be really beneficial both for exposure and focus.

This is a proposal to create a new working group that would be a sub working group of Eventing and would report a brief status summary to the Eventing WG being a sub working group.

### Mission Statement

The Eventing Kafka working group is the innovation forum for Kafka experts, end users and vendors wanting to leverage Kafka for Knative Eventing components. The working group enables a diversity of Kafka based components while ensuring Eventing users end up with a smooth UX for choosing, running and maintaining those components and that Eventing Kafka contributors are algined on a set of working group standards.

### Goals

1. Align the future work across all Kafka based eventing components into a single Kafka focused roadmap.
2. Promote eventing Kafka to the community and attract more Kafka end users, adopters, and contributors.
3. Narrow-focus on Kafka features, Kafka docs and Kafka UX and contribEx. 

### In Scope

* Maintain a roadmap that covers all the Kafka based projects within knative-sandbox.
* Funnel the feedback of Kafka eventing components and surface that to other working groups (e.g. to unblock a specific feature development or enhance the user expeience of Kafka based Eventing components)
* Work with the Docs WG to maintain and enhance the Kafka components docs.
* Maintain a guide that helps Eventing users understand the differences between existing Kafka components and how to choose between them.
* Strive to standardize the APIs across the different Kafka eventing components.
* Unifying and facilitating setting up Kafka CI infrastructure for the current and future Kafka based components.
* Help our contributors 

### Out of Scope

* Dictate any implementation details, architecture, languague or framework on Kafka components.
* Mandate a feature parity between different implementations of Kafka eventing components

**Archived Proposal** https://hackmd.io/@devguyio/kafkawg

**Comment by devguyio** [comment]
@knative/technical-oversight-committee this is still a WIP and any early feedback is welcomed.

**Comment by rhuss** [comment]
@devguyio do you prefer comments in this issue or on the document itself ?

**Comment by devguyio** [comment]
@evankanderson I responded to your comments, thanks for feedback! 

@rhuss whatever you prefer, maybe if the comments are on some proposal specifics then on the document. If they're generic summaries, opinions, then here.

**Comment by evankanderson** [comment]
+1 from @julz and @markusthoemmes 

+1 from @evankanderson with the following addendum:

* For Kafka-motivated changes to API (for example, to enable ordered delivery), the changes should be driven through the cross-technology working groups (e.g. event delivery WG in the case of ordered delivery).

As mentioned in the meeting today, this is our first technology-specific WG; let's see how this goes!

(Approved)

**Comment by devguyio** [comment]
WG folder created: [Folder](https://drive.google.com/drive/folders/1ra5czyKOaMsWuwWXzkhce6_g6rFBdCxg)
WG meeting notes created: [Notes](https://docs.google.com/document/d/1ykBVPtiosGoDjzBklMt9HxbinGrMod6itOvfe6DeVmA/edit)
Calendar invitation created: [Calendar](https://calendar.google.com/event?action=TEMPLATE&tmeid=M2wwMjZ2ZW5lN2lnNXJldXZjNzVrbmkyYTlfMjAyMTA4MjRUMTUzMDAwWiBrbmF0aXZlLnRlYW1fOXE4M2JnMDdxczViOXJyc2xwNWpvcjRsNnNAZw&tmsrc=knative.team_9q83bg07qs5b9rrslp5jor4l6s%40group.calendar.google.com&scp=ALL)

**Comment by evankanderson** [comment]
This is approved and the PR got submitted (sorry about the logjam -- it seems like Peribolos doesn't have a `--dry-run` or `--test` mode.)

/close
